embassy-stm32/src/usart/mod.rs

```diff
@@ -136,6 +136,40 @@ pub enum HalfDuplexReadback {
     Readback,
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+/// Half duplex IO mode
+pub enum OutputConfig {
+    /// Push pull allows for faster baudrates, no internal pullup
+    PushPull,
+    #[cfg(not(gpio_v1))]
+    /// Push pull with internal pull up resistor
+    PushPullPullUp,
```

Comment:
PushPullPullUp and PushPullPullDown make no sense. If the MCU is always driving the pin the pull resistor has no effect (other than wasting current).